Add `transaction.get` route.

andrew-chang-dewitt opened the pull request:
Closes #1

andrew-chang-dewitt committed:
Add `transaction.get` route.

Closes #1

andrew-chang-dewitt committed:
Improve test coverage of single item get.

andrew-chang-dewitt committed:
Merge branch 'main' into add-route-get

Pull changes to CI scripts.

andrew-chang-dewitt committed:
Fix relative imports so mypy can understand.

Mypy threw errors related to imports when typchecking; changing to
relative imports fixed errors.

andrew-chang-dewitt committed:
Change start command to run server as module.

Required because of change in import structure to relative.

andrew-chang-dewitt committed:
Fix build script compose command arguments.

andrew-chang-dewitt committed:
Apply fixes per linters.

andrew-chang-dewitt committed:
Remove unit testing on Model.

These tests just check if the SQL query is composed correctly, but they
fail to actually predict real output. By using a helper method to build
the query since it is impossible to use psycopg2's query building
without a db connection, we're actually just testing our helper query.

Instead, by using thorough integration tests, we'll both check that our
queries are built correctly & that they have the desired effect on the
database.

andrew-chang-dewitt committed:
Add max count behavior on get route.

CrackerJack625 commented on test/unit/helpers/database.py:
```diff
@@ -0,0 +1,62 @@
+        # print(f'composing {q}')
+        strings: List[str] = []
+
+        for p in q.seq:
```
Just a suggestion, something someone who has reviewed my code mentioned to always detail out these to say what they are doing. This will help with code review down the road. Alphabet salad becomes hard to manage down the road.

andrew-chang-dewitt replied:
Resolved in [commit a22ce37](https://github.com/cheese-drawer/transaction-service/pull/7/commits/a22ce372fb223e66ce1caeb7a0299276f846ccb5).

CrackerJack625 commented on test/integration/test_request_response_api.py:
```diff
@@ -43,195 +50,168 @@ def tearDownModule() -> None:
+        next_ten_ids = [self.trim_msg_id(tran['_id']) for tran in next_ten]
 
-        conn = psycopg2.connect('postgres://test:pass@localhost/dev')
+        print(next_ten_ids)
```
Do we want to remove our print lines before PR reviews? If so maybe it needs to be moved over to logging instead.

andrew-chang-dewitt replied:
I think this ones fine since it’s in a test and helpful for debugging, but in source code I would definitely say we should remove all print statements or replace them with logging calls.

andrew-chang-dewitt committed:
Pin SQLAlchemy version to 1.3.x.

Solves errors with migra.

andrew-chang-dewitt committed:
Debug CI errors.

andrew-chang-dewitt committed:
Pin sqlbag version, debugging CI

andrew-chang-dewitt committed:
Pin sqlbag version, debugging CI.

andrew-chang-dewitt committed:
Pin sqlbag version, debugging CI.

andrew-chang-dewitt committed:
Resolve review, improve variable names.

andrew-chang-dewitt commented:
Needs integration testing errors resolved before reviewing again. Should be route in JSON encoding, will try pulling in updates from add-route-update that improve encoding.

andrew-chang-dewitt committed:
Fix route Response JSON encoding.

Update amqp_worker to 0.2.0 to get new features related to JSON
encoding. Implement extended encoder & custom pattern factory, then pass
to worker on initilization.

andrew-chang-dewitt committed:
Fix intermittent duplicate ID issues.

Create UUID values with uuid1 instead of uuid4 in testing to prevent id
collisions.
